src/app/services/date-select.service.ts

```diff
@@ -11,22 +12,28 @@ export class DateSelectService {
 
   constructor() {
     // TODO - Update once date selector component is created
-    const yesterdaysDate = new Date(Date.now());
-    yesterdaysDate.setUTCDate(yesterdaysDate.getDate() - 1)
-    const twoDaysBeforeDate = new Date(yesterdaysDate);
-    twoDaysBeforeDate.setUTCDate(yesterdaysDate.getDate() - 3);
+    const yesterdaysDate = moment().subtract(1, 'd');
+    const threeDaysBeforeTodayDate = moment().subtract(3, 'd');
 
     this.dateRangeSelected = {
       to: yesterdaysDate,
-      from: twoDaysBeforeDate
+      from: threeDaysBeforeTodayDate
     }
     
     this.subscription = new Subject<DateRange>();
   }
 
-  setDateRange(from: Date, to: Date): void {
-    this.dateRangeSelected = { from, to };
-    this.subscription.next(this.dateRangeSelected);
+  setDateRange(from: moment.Moment, to: moment.Moment): void {
+    if (this.validateDates(from, to)) {
```

Comment:
It might make sense for the service to re-send its current values when given an invalid input.
That way all components will always have valid input (currently, I can manually input e.g. a starting date which is in the future and it will stay that way until manually corrected).

Reply:
That does make sense, only thing is that I think it'll prompt a request to be sent by anything subscribe to the observable like the historical api stuff. But that'll be sorted once we have caching involved although it may cause flickering of some components till that happens.

I've made some changes so it'll revert changes if the range is bad (if start date > end date) but need to figure out a way to make it do the same for invalid dates (eg 50/9/2020). Issue is that the input doesn't register 50/9/2020 as a date change because its an invalid date, even though it'll take 300/9/2020 (as 30/9/2020). Must be something to do with how the form works, just couldn't figure it out this morning.